(Rebased) AssetRipper.CLI

YoshiRulz opened the pull request:
<!--TODO set up CI
[![dev build for branch | YoshiRulz:cli-rebased](https://img.shields.io/badge/dev_build_for_branch-YoshiRulz:cli--rebased-8250DF?logo=github&logoColor=333333&style=popout)](https://nightly.link/YoshiRulz/AssetRipper/workflows/TODO/cli-rebased?preview)
-->

Simply exposes the "Export Unity Project" and "Export Primary Content" functions of the web interface via a new command-line interface:
```
$> AssetRipper.CLI unityproject "/path/to/game" "/target/output/path"
$> # and
$> AssetRipper.CLI primarycontent "/path/to/game" "/target/output/path"
```
You should also be able to compile it for macOS or Windows, though I don't use those OSes.
edit: Also available [in Nixpkgs](https://github.com/YoshiRulz/nixpkgs/tree/assetripper-cli), sadly as a fork and not an overlay.

<!--
The upstream maintainer has shot down requests for a CLI, see my earlier PR below for details.
Personally, their attitude is pushing me to consider alternative tools, but for now I'm stuck with it.
-->

Emptiest commented:
Hi @YoshiRulz for the uninitiated, how would one go about compiling this on Linux? Any help would be appreciated!

YoshiRulz commented:
The standard way: `dotnet publish -c Release -r linux-x64`

YoshiRulz committed:
Allow for deterministic values of `timeCreated` in output via `SOURCE_DATE_EPOCH` environmental variable (#2056)

YoshiRulz committed:
Add CLI

change to solution file might be wrong; was done with `dotnet sln` +
hand-fixes
also some of the MSBuild props might be specific to ASP.NET and thus
useless here
